pkg/apis/azuredisk/v1beta1/types.go

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// AzVolume is a specification for a AzVolume resource
+type AzVolume struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   AzVolumeSpec   `json:"spec"`
+	Status AzVolumeStatus `json:"status"`
+}
+
+// AzVolumeSpec is the spec for a AzVolume resource
+type AzVolumeSpec struct {
+	UnderlyingVolume string `json:"underlyingVolume"`
+}
+
+// AzVolumeStatus is the status for a AzVolume resource
+type AzVolumeStatus struct {
+	UnderlyingVolume int32 `json:"underlyingVolume"`
```

Comment:
Shouldn't this field be `State`?

Reply:
Using this PR as a skeleton for generating CRDs. Will add the fields as a separate PR.